### PR TITLE
Add support for sections in the doc

### LINKED ISSUE #53: Add support for sections in the doc
The ability to add sections to group together functions.

For example if a script looks like:
```
  # @section Section A
  # @description a
  # a
  
  # @description func a
  # line a 2
  a() {
  }
  
  # @section Section B
  # @description b
  
  # @description func b
  # ab
  b() {
  }
```

the generated md file would look like:
```
## Section A

a
a

### a

func a
line a 2

## Section B

b

### b

func b
ab

```

### PR DESCRIPTION
**Pull request by Ariana-Hlavaty-i2**
Feature request implementation. Resolves #53

### CODE REVIEW

File: shdoc [hunk 1]
```diff
@@ -185,7 +191,19 @@ function render_docblock(func_name, description, docblock) {
     debug("→ → func_name: [" func_name "]")
     debug("→ → description: [" description "]")
 
-    lines[0] = render("h3", func_name)
+    if (section != "") {
+      lines[0]=render("h2", section)
```
Comment (kovetskiy) [inline shdoc:195]: Could you add some spaces around `=` here?

**Review by kovetskiy — CHANGES_REQUESTED** [review]
Hi!

Thank you for your contribution.

A few nitpicks and we can merge the PR.

**Review by Ariana-Hlavaty-i2 — COMMENTED** [review]
Required spaces added as per comments

**Comment by kovetskiy** [comment]
Thanks!

**Pull request merged**